tokio-trace-macros/example/src/main.rs

```diff
@@ -0,0 +1,28 @@
+#[macro_use]
+extern crate tokio_trace;
+#[macro_use]
+extern crate tokio_trace_macros;
+use tokio_trace::Level;
+
+#[path = "../../../tokio-trace/examples/sloggish/sloggish_subscriber.rs"]
+mod sloggish;
+use self::sloggish::SloggishSubscriber;
+
+fn main() {
+    tokio_trace::Dispatcher::builder()
+        .add_subscriber(SloggishSubscriber::new(2))
+        .init();
+
+    let server_span = span!("server", local_port = 8888);
+    server_span.clone().enter(move || {
+        suggest_band("Parquet Courts");
+    });
+}
+
+#[trace]
+#[deprecated]
```

Comment:
Why is this tagged as `deprecated`? I'd prefer not to see the warning when I `cargo build --all`.

I'm guessing this is to test that the proc-macro doesn't break when the annotated functions have other attributes? It might be good to use something like `#[inline]` --- which doesn't raise a warning --- instead?

Reply:
> I'm guessing this is to test that the proc-macro doesn't break when the annotated functions have other attributes? It might be good to use something like #[inline] --- which doesn't raise a warning --- instead?

Yeah, that's the case. It was to test that other attributes wouldn't be dropped. Do you know of a good mechanism to test proc macros (or macros in general?)

Reply:
I think that with macros in general, my experience tends to be that bugs in the macro usually results in a compile error. So, just trying to compile the examples (which `cargo test --all` does) should be sufficient.

You could probably also write unit tests for the output of the proc-macro function if you wanted. You could probably make some assertions about the output `TokenStream`?